src/components/Carousel.tsx

```diff
@@ -1,26 +1,95 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li>
-        <img src="./img/1.png" alt="1" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="2" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="3" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="4" />
-      </li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type Props = {
+  images: string[];
+  step: number;
+  frameSize: number;
+  itemWidth: number;
+  animationDuration: number;
+  infinite: boolean;
+};
+
+const Carousel: React.FC<Props> = ({
+  images,
+  step,
+  frameSize,
+  itemWidth,
+  animationDuration,
+  infinite,
+}) => {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const handleNext = () => {
+    if (currentIndex < images.length - step) {
+      setCurrentIndex(currentIndex + step);
+    } else if (infinite) {
+      setCurrentIndex((currentIndex + step) % images.length);
+    }
+  };
+
+  const handlePrev = () => {
+    if (currentIndex > 0) {
+      setCurrentIndex(currentIndex - step);
+    } else if (infinite) {
+      setCurrentIndex(images.length - step);
+    }
+  };
+
+  useEffect(() => {
+    const maxIndex = images.length - frameSize;
+
+    if (currentIndex > maxIndex) {
+      setCurrentIndex(maxIndex);
+    }
+  }, [frameSize, images.length, currentIndex]);
+
+  return (
+    <div className="Carousel" style={{ width: `${itemWidth * frameSize}px` }}>
+      <ul
+        className="Carousel__list"
+        style={{
+          width: `${frameSize * itemWidth}px`,
+          transform: `translateX(-${currentIndex * itemWidth}px)`,
+          transition: `transform ${animationDuration}ms ease-in-out`,
+        }}
+      >
+        {images.map(image => (
+          <li>
+            <img
+              style={{
+                width: `${itemWidth}px`,
+              }}
+              key={image}
```

Comment:
```suggestion
          <li key={image}>
            <img
              style={{
                width: `${itemWidth}px`,
              }}
```
<img width="646" alt="image" src="https://github.com/mate-academy/react_carousel/assets/70220667/f6ce8604-bf1f-4d7c-a47f-315e3f8fd768">